### ISSUE TITLE
Menu with icon - Accessibility violations "Elements must only use allowed ARIA attributes"

### ISSUE THREAD

**Issue by ZhuoyuJin** [Sage/carbon#4986]
### Current behaviour

One accessibility violation in "with icon" story for Menu componen
![Image 22-03-2022 at 12 37](https://user-images.githubusercontent.com/72838192/159483684-b5ff0cc4-edbf-4a7b-8ad6-421b4a486a5e.jpg)
t

### Expected behaviour

No Accessibility violations

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/menu--with-icon

### Carbon Version

latest

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by nicktitchmarsh** [comment]
FE-5090

**Comment by Chris3y** [comment]
You may be aware but just to add if not, we see the exact same errors from icons inside fields (validation).
